src/zeep/xsd/types/complex.py

```diff
@@ -431,14 +432,20 @@ def restrict(self, base):
             is_global=self.is_global)
         return new.resolve()
 
-    def signature(self, schema=None, standalone=True):
+    def signature(self, schema=None, standalone=True, path=None):
         parts = []
+
+        path = utils.extend_path(path, self)
+
         for name, element in self.elements_nested:
-            part = element.signature(schema, standalone=False)
+            if element in path:
+                part = '%s(...)' % name
```

Comment:
Using a list is not effective. We could use a set instead of the list. Sadly, path elements are not hashable.